src/components/Avatar.js

```diff
@@ -68,29 +77,39 @@ class Avatar extends PureComponent {
             return null;
         }
 
+        const isWorkspace = this.props.type === CONST.ICON_TYPE_WORKSPACE;
+        const iconSize = StyleUtils.getAvatarSize(this.props.size);
+
         const imageStyle = [
             StyleUtils.getAvatarStyle(this.props.size),
             ...this.props.imageStyles,
+            StyleUtils.getAvatarBorderRadius(this.props.size, this.props.type),
         ];
 
         const iconStyle = [
             StyleUtils.getAvatarStyle(this.props.size),
             styles.bgTransparent,
             ...this.props.imageStyles,
         ];
-        const iconSize = StyleUtils.getAvatarSize(this.props.size);
+
+        const iconFillColor = isWorkspace ? StyleUtils.getDefaultWorspaceAvatarColor(this.props.name).fill : this.props.fill;
+        const fallbackAvatar = isWorkspace ? ReportUtils.getDefaultWorkspaceAvatar(this.props.name) : this.props.fallbackIcon;
```

Comment:
Same as above, I think we can add in the documentation that `this.props.fallbackIcon` is ignored for `this.props.type === CONST.ICON_TYPE_WORKSPACE`

Reply:
Is it better practice to have Avatar handle this calculation for Fallback avatar `ReportUtils.getDefaultWorkspaceAvatar(this.props.name)`
 or should we do this calculation and pass it in as a prop when we're building these workspace avatars? 

i.e in `WorkspaceInitialPage` do:
``` js
                                                <Avatar
                                                    containerStyles={styles.avatarLarge}
                                                    imageStyles={[styles.avatarLarge, styles.alignSelfCenter]}
                                                    source={this.props.policy.avatar ? this.props.policy.avatar : ReportUtils.getDefaultWorkspaceAvatar(policyName)}
                                                    fallbackIcon={ReportUtils.getDefaultWorkspaceAvatar(policyName)}
                                                    size={CONST.AVATAR_SIZE.LARGE}
                                                    name={policyName}
                                                    type={CONST.ICON_TYPE_WORKSPACE}
                                                />
```

Reply:
So then you don't need to pass the `name` to `Avatar` and the dependency between the props is removed?

Reply:
I thought that a possibility of calculating the `iconFillColor` and `fallbackAvatar` outside, so you don't have this thing that "a prop is ignore based on the value of another"... but honestly I think It is not worth it... you would have to repeat that code in many places you use `<Avatar>` I think.

I would just go with adding a small comment to the props, and leave like you are doing now

Reply:
Yeah DRY was my intention as well! I'll stick with it and add the comment